src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -119332,5 +119453,182 @@ class C<T>
                 //         P6 = t; // 12
                 Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "t").WithLocation(18, 14));
         }
+
+        [Fact]
+        [WorkItem(39922, "https://github.com/dotnet/roslyn/issues/39922")]
+        public void MaybeNullT_26()
+        {
+            var source =
+@"#nullable enable
+using System.Diagnostics.CodeAnalysis;
+class Program
+{
+    static void F1<T>(   [AllowNull]T x,               T y) { y = x; }
+    static void F2<T>(   [AllowNull]T x,    [AllowNull]T y) { y = x; }
+    static void F3<T>(   [AllowNull]T x, [DisallowNull]T y) { y = x; }
```

Comment:
Why is there no warning on this line?

Reply:
Ohh, I see, because these aren't observable from the outer scope. Never mind.

---
In reply to: [349839731](https://github.com/dotnet/roslyn/pull/39983#discussion_r349839731) [](ancestors = 349839731)